tests/verify.go

```diff
@@ -2,76 +2,54 @@ package tests
 
 import (
 	"net/http"
-	"os"
+	"path"
 	"testing"
 	"time"
 )
 
 func invoke(t *testing.T, name string, query string, expectedStatusCode ...int) []byte {
+	t.Helper()
 	return invokeWithVerb(t, http.MethodPost, name, query, expectedStatusCode...)
 }
 
 func invokeWithVerb(t *testing.T, verb string, name string, query string, expectedStatusCode ...int) []byte {
+	t.Helper()
+
 	attempts := 30 // i.e. 30x2s = 1m
 	delay := time.Millisecond * 750
 
 	breakoutStatus := []int{http.StatusUnauthorized}
 
-	uri := os.Getenv("gateway_url") + "function/" + name
-	if len(query) > 0 {
-		uri = uri + "?" + query
-	}
+	uri := gatewayUrl(t, path.Join("function", name), query)
 
+	var bytesOut []byte
 	for i := 0; i < attempts; i++ {
 
-		bytesOut, res, err := httpReq(uri, verb, nil)
-
-		if err != nil {
-			t.Log(err.Error())
-			t.Fail()
-		}
+		bytesOut, res := request(t, uri, verb, nil)
 
-		validMatch := false
 		for _, code := range expectedStatusCode {
 			if res.StatusCode == code {
-				validMatch = true
-				break
+				// success, we can stop now
+				t.Logf("[%d/%d] Got correct response: %v - %s", i+1, attempts, res.StatusCode, uri)
+				return bytesOut
 			}
 		}
 
-		breakout := false
-
+		// handle fatal errors that we can not retry
 		for _, code := range breakoutStatus {
 			if res.StatusCode == code {
-				breakout = true
-				break
+				t.Fatalf("Received breakout-status %d, invoke failed with: %s", res.StatusCode, bytesOut)
```

Comment:
I'm not sure why t.Fatal is better here (mid-loop) than after breaking out? some sort of function like `matchStatus` could be extracted perhaps, so that single-responsibility could be maintained, just checking the status vs. checking status and calling Fatal.